public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -83,18 +82,27 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
   ): Observable<DataQueryResponse> => {
     const validLogQueries = logQueries.filter(this.filterQuery);
 
-    const startQueryRequests: StartQueryRequest[] = validLogQueries.map((target: CloudWatchLogsQuery) => ({
-      queryString: target.expression || '',
-      refId: target.refId,
-      logGroupNames: target.logGroupNames || this.instanceSettings.jsonData.defaultLogGroups || [],
-      logGroups: target.logGroups || this.instanceSettings.jsonData.logGroups,
-      region: super.replaceVariableAndDisplayWarningIfMulti(
-        this.getActualRegion(target.region),
-        options.scopedVars,
-        true,
-        'region'
-      ),
-    }));
+    const startQueryRequests: StartQueryRequest[] = validLogQueries.map((target: CloudWatchLogsQuery) => {
+      const logGroupArns = interpolateStringArrayUsingSingleOrMultiValuedVariable(
+        this.templateSrv,
+        (target.logGroups || this.instanceSettings.jsonData.logGroups || []).map((lg) => lg.arn)
+      );
+
+      // need to support legacy format variables too
+      const logGroupNames = interpolateStringArrayUsingSingleOrMultiValuedVariable(
+        this.templateSrv,
+        target.logGroupNames || this.instanceSettings.jsonData.defaultLogGroups || [],
+        'text'
+      );
+
+      return {
+        refId: target.refId,
+        region: this.templateSrv.replace(this.getActualRegion(target.region)),
+        queryString: this.templateSrv.replace(target.expression || ''),
+        logGroups: uniq(logGroupArns).map((arn) => ({ arn, name: arn })),
+        logGroupNames: uniq(logGroupNames),
```

Comment:
are these uniqs necessary?

Reply:
Yes. Let's say for example this is the logGroupNames array:
```typescript 
['logGroup1', '$logGroupVariable']
```
If $logGroupVariable is a multi-valued variable, and one of it's value is the same as `logGroup1` which is already in the logGroupNames array, the array will include dupes:
```typescript 
// expanded
['logGroup1', 'logGroup1', 'logGroup2']
```
When logGroupNames includes dupes, the  `StartLogQuery` api will return an error. Weirdly, it returns a permission error (?). However, removing dupes with the uniq func solves the problem.